File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java

class Solution {
    public int solution(String s) {
        int answer = 0;
        
        String[] keys = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        Integer[] values = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
        
        for(int i = 0; i < keys.length; i++) {
            if(s.contains(keys[i])) {
                s = s.replace(keys[i], String.valueOf(values[i]));
            }
        }
        
        answer = Integer.valueOf(s);
        
        return answer;
    }
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
import re

def solution(s: str):
    arrMap = {
        'zero':'0', 'one':'1', 'two':'2', 'three':'3', 'four':'4',
        'five':'5', 'six':'6', 'seven':'7', 'eight':'8', 'nine':'9'
    }

    target = '|'.join(arrMap.keys())

    answer = re.sub(rf'{target}', lambda m: arrMap[m.group()], s)

    return int(answer) # 숫자로 반환할 경우